src/ui/ActionFilters/McExceptionFilter.cs

```diff
@@ -9,11 +9,20 @@ public class McExceptionFilter : IExceptionFilter
     {
         public void OnException(ExceptionContext context)
         {
-            if (context.Exception is SwaggerException swaggerException
-                && swaggerException.StatusCode == (int)HttpStatusCode.Unauthorized)
+            if (context.Exception is SwaggerException swaggerException)
             {
-                context.Result = new RedirectToActionResult("Index", "Error", new { statusCode = 401 });
-                context.ExceptionHandled = true;
+                if (swaggerException.StatusCode == (int) HttpStatusCode.Unauthorized)
+                {
+                    context.Result = new RedirectToActionResult("Index", "Error", new { statusCode = 401 });
+                    context.ExceptionHandled = true;
+                }
+
+                var unavailableForLegalReasons = 451; // core 2.1 only (int) HttpStatusCode.UnavailableForLegalReasons
+                if (swaggerException.StatusCode == unavailableForLegalReasons)
+                {
+                    context.Result = new RedirectToActionResult("AcceptTerms", "Home", null);
+                    context.ExceptionHandled = true;
+                }
```

Comment:
use a switch case statement instead of two ifs

minor

Reply:
rule of 3 stills applies, not dealing with it til then